### PR TITLE
[block] move `VerifyDeltaStateDigest`, `VerifyReceiptRoot`, and `VerifyTransactionRoot` to `header.go`

### PR DESCRIPTION
**Pull request by Liuhaai**

### CODE REVIEW

File: blockchain/block/block.go [hunk 1]
```diff
@@ -88,37 +79,21 @@ func (b *Block) Deserialize(buf []byte) error {
 	b.Receipts = nil
 
 	// verify merkle root can match after deserialize
-	txHash, err := b.CalculateTxRoot()
-	if err != nil {
-		log.L().Debug("error in getting hash ", zap.Error(err))
-		return err
-	}
-	if err := b.VerifyTxRoot(txHash); err != nil {
+	if err := b.VerifyTxRoot(); err != nil {
```
Comment (dustinxie) [inline blockchain/block/block.go:82]: good clean-up

File: blockchain/block/block.go [hunk 2]
```diff
@@ -88,37 +79,21 @@ func (b *Block) Deserialize(buf []byte) error {
...
 	}
-	return nil
-}
-
-// VerifyReceiptRoot verifies the receipt root in header
-func (b *Block) VerifyReceiptRoot(root hash.Hash256) error {
-	if b.Header.receiptRoot != root {
-		return ErrReceiptRootMismatch
+	if err := b.Header.VerifyTransactionRoot(root); err != nil {
+		return err
```
Comment (dustinxie) [inline blockchain/block/block.go:96]: `return b.Header.VerifyTransactionRoot(root)`

File: blockchain/block/utils.go [hunk 3]
```diff
@@ -47,29 +45,3 @@ func calculateTransferAmount(acts []action.SealedEnvelope) *big.Int {
...
-// VerifyBlock verifies the block signature and tx root
-func VerifyBlock(blk *Block) error {
-	// verify new block's signature is correct
-	if !blk.VerifySignature() {
-		return errors.Errorf(
-			"failed to verify block's signature with public key: %x",
-			blk.PublicKey(),
-		)
-	}
-
```
Comment (dustinxie) [inline blockchain/block/utils.go:60]: we can still keep this func, just replace with bottom part with `b.VerifyTxRoot()`

File: state/factory/workingset.go [hunk 4]
```diff
@@ -514,10 +514,10 @@ func (ws *workingSet) ValidateBlock(ctx context.Context, blk *block.Block) error
 	if err != nil {
 		return err
 	}
-	if err = blk.VerifyDeltaStateDigest(digest); err != nil {
+	if err = blk.Header.VerifyDeltaStateDigest(digest); err != nil {
```
Comment (CoderZhi) [inline state/factory/workingset.go:517]: revert

File: state/factory/workingset.go [hunk 5]
```diff
@@ -514,10 +514,10 @@ func (ws *workingSet) ValidateBlock(ctx context.Context, blk *block.Block) error
...
 		return errors.Wrap(err, "failed to verify delta state digest")
 	}
-	if err = blk.VerifyReceiptRoot(calculateReceiptRoot(ws.receipts)); err != nil {
+	if err = blk.Header.VerifyReceiptRoot(calculateReceiptRoot(ws.receipts)); err != nil {
```
Comment (CoderZhi) [inline state/factory/workingset.go:520]: revert

File: blockchain/block/header.go [hunk 6]
```diff
@@ -174,6 +182,30 @@ func (h *Header) VerifySignature() bool {
...
+	return nil
+}
+
+// VerifyDeltaStateDigest verifies the delta state digest in header
+func (h *Header) VerifyTransactionRoot(root hash.Hash256) error {
+	if h.txRoot != root {
+		return ErrTxRootMismatch
+	}
+	return nil
+}
```
Comment (CoderZhi) [inline blockchain/block/header.go:207]: return bool for all these three functions

File: blockchain/block/utils.go (hunk 3 above)
Comment (Liuhaai) [reply blockchain/block/utils.go:60]: This func name is unclear. It doesn't verify blk entirely

Comment (Liuhaai) [reply blockchain/block/utils.go:60]: After discussion, it is renamed to `VerifySignatureAndTxRoot`

File: blockchain/block/block.go (hunk 1 above)
Comment (CoderZhi) [inline blockchain/block/block.go:82]: return b.VerifyTxRoot()

File: blockchain/block/block.go [hunk 7]
```diff
@@ -88,37 +79,21 @@ func (b *Block) Deserialize(buf []byte) error {
...
-}
-
-// VerifyDeltaStateDigest verifies the delta state digest in header
-func (b *Block) VerifyDeltaStateDigest(digest hash.Hash256) error {
-	if b.Header.deltaStateDigest != digest {
-		return ErrDeltaStateMismatch
+func (b *Block) VerifyTxRoot() error {
+	root, err := b.CalculateTxRoot()
+	if err != nil {
+		log.L().Debug("error in getting hash ", zap.Error(err))
```
Comment (CoderZhi) [inline blockchain/block/block.go:92]: delete extra suffix space

File: blockchain/block/utils.go (hunk 3 above)
Comment (CoderZhi) [reply blockchain/block/utils.go:60]: can you explain why we need to create this function for block? It doesn't help at all.

File: blockchain/block/block.go (hunk 1 above)
Comment (dustinxie) [reply blockchain/block/block.go:82]: +1

File: blockchain/block/block.go (hunk 7 above)
Comment (dustinxie) [reply blockchain/block/block.go:92]: this func also used in `blockstore.go`, update there too

File: blockchain/block/utils.go (hunk 3 above)
Comment (Liuhaai) [reply blockchain/block/utils.go:60]: revert

**Review by dustinxie — APPROVED** [review]

File: blockchain/block/block.go (hunk 1 above)
Comment (dustinxie) [reply blockchain/block/block.go]: fixed in latest commit

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**